src/main/scala/uk/gov/nationalarchives/tdr/api/db/repository/ConsignmentRepository.scala

```diff
@@ -20,6 +23,20 @@ class ConsignmentRepository(db: Database) {
     db.run(insertQuery += consignmentRow)
   }
 
+  def updateExportLocation(exportLocationInput: ConsignmentFields.UpdateExportLocationInput, timestamp: Timestamp): Future[Int] = {
+    val update = Consignment.filter(_.consignmentid === exportLocationInput.consignmentId)
+      .map(c => (c.exportlocation, c.exportdatetime))
+      .update((Option(exportLocationInput.exportLocation), Option(timestamp)))
+    db.run(update)
+  }
+
+  def updateTransferInitiated(consignmentId: UUID, userId: Option[UUID], timestamp: Timestamp): Future[Int] = {
```

Comment:
The [authoriser](https://github.com/nationalarchives/tdr-consignment-api/blob/master/src/main/scala/uk/gov/nationalarchives/tdr/api/http/Routes.scala#L31) just checks to see if it's a valid keycloak token which it is without the user id parameter. We could change the authoriser to make sure there is one but for now, we don't.

Reply:
Ah, that's interesting. Can you remember if the user ID is None when the token comes from a service client?

If service client tokens have a user ID, maybe we should reject it in the authoriser? If they _don't_ have a user ID, maybe we should reject requests where we have to keep an audit trail of the user?

Reply:
I've changed the auth-utils `userId` method so it returns a `UUID` instead of `Option[String]` The method which gets the token will return a failure if the userId isn't set so this is fine.